### PR TITLE
Add simple consistency proof integration test 

### PR DESCRIPTION
**Pull request by Martin2112**
And switch to printing everything in hex not base64 when it doesn't match. Include the whole proof when reporting mismatches so it can be examined. Implements #212 .

### CODE REVIEW

File: integration/log_integration_test.go [hunk 1]
```diff
@@ -45,9 +46,23 @@ type testParameters struct {
 	sequencingPollWait  time.Duration
 }
 
+type consistencyProofParams struct {
+	firstTreeSize  int64
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: (style) Go tends towards terse names, so the type and field names here seem a bit long for something that's only used in this file.

File: integration/log_integration_test.go [hunk 2]
```diff
@@ -372,23 +404,35 @@ func checkInclusionProofsAtIndex(index int64, logID int64, tree *merkle.InMemory
...
 		}
 	}
 
 	return nil
 }
 
+func checkConsistencyProof(consistParams consistencyProofParams, treeID int64, tree *merkle.InMemoryMerkleTree, client trillian.TrillianLogClient, params testParameters) error {
+	// We expect the proof request to succeed
+	ctx, cancel := getRPCDeadlineContext()
+	resp, err := client.GetConsistencyProof(ctx, &trillian.GetConsistencyProofRequest{LogId: treeID,
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: (style) new line for LogId, new line for }

File: integration/log_integration_test.go [hunk 3]
```diff
@@ -45,9 +46,23 @@ type testParameters struct {
...
 
+type consistencyProofParams struct {
+	firstTreeSize  int64
+	secondTreeSize int64
+}
+
 // inclusionProofTestIndices are the 0 based leaf indices to probe inclusion proofs at.
 var inclusionProofTestIndices = []int64{5, 27, 31, 80, 91}
 
+// consistencyProofTestParams are the intervals (to be multiplied by the batch size)
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: The "mulitplied by batch size" comment belongs with the structure definition

File: integration/log_integration_test.go [hunk 4]
```diff
@@ -452,3 +496,43 @@ func min(a, b int64) int64 {
 
 	return b
 }
+
+// compareLogAndTreeProof compares a proof received from the log against one generated by
+// an in memory merkle tree. It ensures the proofs contain an identical list of node hashes.
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: merkle -> Merkle

File: integration/log_integration_test.go [hunk 5]
```diff
@@ -143,6 +158,23 @@ func TestLogIntegration(t *testing.T) {
 			t.Fatalf("Log inclusion index: %d proof checks failed: %v", testIndex, err)
 		}
 	}
+
+	// Step 6 - Test some consistency proofs
+
+	// Make some proof requests that we know should not succeed
+	for _, consistParams := range consistencyProofBadTestParams {
+		if err := checkConsistencyProof(consistParams, treeID, tree, client, params); err == nil {
+			t.Fatalf("Log consistency for %v: unexpected proof returned", consistParams)
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: (pre-existing/consistency) These should be `Errorf` not `Fatalf`.

File: integration/log_integration_test.go [hunk 6]
```diff
@@ -452,3 +496,43 @@ func min(a, b int64) int64 {
 
 	return b
 }
+
+// compareLogAndTreeProof compares a proof received from the log against one generated by
+// an in memory merkle tree. It ensures the proofs contain an identical list of node hashes.
+func compareLogAndTreeProof(logProof *trillian.Proof, treeProof []merkle.TreeEntryDescriptor) error {
+	// Compare the proof lengths
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: Idle thought: would it be simpler replace this fn with a comparison of the results of `formatLogProof` and `formatTreeProof` ?

File: integration/log_integration_test.go (hunk 1 above)
Comment (Martin2112) [inline integration/log_integration_test.go]: Done.  I'm overall unimpressed with Go style and the seeming need to make it differ from every other language. They can get off my lawn.

File: integration/log_integration_test.go (hunk 3 above)
Comment (Martin2112) [inline integration/log_integration_test.go]: Done.

File: integration/log_integration_test.go (hunk 5 above)
Comment (Martin2112) [inline integration/log_integration_test.go]: Done.

File: integration/log_integration_test.go (hunk 2 above)
Comment (Martin2112) [inline integration/log_integration_test.go]: Done, but it looks wrong to me.

File: integration/log_integration_test.go (hunk 4 above)
Comment (Martin2112) [inline integration/log_integration_test.go]: Done + a couple of others.

File: integration/log_integration_test.go [hunk 7]
```diff
@@ -417,17 +420,19 @@ func checkInclusionProofsAtIndex(index int64, logID int64, tree *merkle.InMemory
 func checkConsistencyProof(consistParams consistencyProofParams, treeID int64, tree *merkle.InMemoryMerkleTree, client trillian.TrillianLogClient, params testParameters) error {
 	// We expect the proof request to succeed
 	ctx, cancel := getRPCDeadlineContext()
-	resp, err := client.GetConsistencyProof(ctx, &trillian.GetConsistencyProofRequest{LogId: treeID,
-		FirstTreeSize:  consistParams.firstTreeSize * int64(params.sequencerBatchSize),
-		SecondTreeSize: (consistParams.secondTreeSize * int64(params.sequencerBatchSize))})
+	resp, err := client.GetConsistencyProof(ctx,
+		&trillian.GetConsistencyProofRequest{LogId: treeID,
```
Comment (daviddrysdale) [inline integration/log_integration_test.go]: (style) Could still do with a new line for LogId.

**Pull request merged**